dlt/normalize/items_normalizers.py

```diff
@@ -112,6 +122,45 @@ def __call__(
 
 
 class ParquetItemsNormalizer(ItemsNormalizer):
+    RECORD_BATCH_SIZE = 1000
+
+    def _write_with_dlt_columns(
+        self, extracted_items_file: str,
+        normalize_storage: NormalizeStorage,
+        load_storage: LoadStorage, load_id: str, schema: Schema, root_table_name: str,
+        add_load_id: bool, add_dlt_id: bool
+    ) -> int:
+        import numpy as np
+        new_columns: List[Any] = []
+        if add_load_id:
+            schema.update_table({"name": root_table_name, "columns": {"_dlt_load_id": {"name": "_dlt_load_id", "data_type": "text"}}})
+            new_columns.append((
+                pyarrow.pyarrow.field("_dlt_load_id", pyarrow.pyarrow.string(), nullable=False),
+                lambda batch: np.full(batch.num_rows, load_id)
+            ))
+        if add_dlt_id:
+            schema.update_table({"name": root_table_name, "columns": {"_dlt_id": {"name": "_dlt_id", "data_type": "text"}}})
+            new_columns.append((
+                pyarrow.pyarrow.field("_dlt_id", pyarrow.pyarrow.string(), nullable=False),
+                lambda batch: (uniq_id_base64(10) for _ in range(batch.num_rows))
```

Comment:
is this what we do in relational normalizer?

Reply:
Looked like it, I didn't look too deeply into this code: https://github.com/dlt-hub/dlt/blob/sthor%2Fnormalize-parquet/dlt/common/normalizers/json/relational.py#L145-L147  Am I missing something?   

Should make a common function to generate ID though.

Reply:
Optimized this too with ids created in bulk with the underlying function: https://github.com/dlt-hub/dlt/pull/704/files#diff-18351bb2bb37a436db3f126f90a273da557aa78cdc1e0c17164cc235a66c764dR43-R51    
The individual `token_bytes` really add up.